src/script/standard.h

```diff
@@ -70,15 +81,49 @@ class CNoDestination {
  */
 typedef boost::variant<CNoDestination, CKeyID, CScriptID> CTxDestination;
 
+/** Get the name of a txnouttype as a C string, or nullptr if unknown. */
 const char* GetTxnOutputType(txnouttype t);
 
+/**
+ * Parse a scriptPubKey and identify script type for standard scripts. Assigns
+ * the result to the typeRet reference parameter and returns true if successful.
+ */
 bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::vector<unsigned char> >& vSolutionsRet);
+
+/**
+ * Parse a standard scriptPubKey for the destination address. Assigns result to
```

Comment:
Happy to add if you prefer, but that seems like something that the implementation might change. Is it possible that a new `CTxDestination` for witness programs will be added? It's also interesting that this function does not return a `CNoDestination` for `TX_NULL_DATA` scripts.